src/com/ableton/VirtualEnv.groovy

```diff
@@ -104,18 +104,13 @@ class VirtualEnv implements Serializable {
 
     this.script = script
 
-    String tempDir = '/tmp'
     if (script.isUnix()) {
       activateSubDir = 'bin'
     } else {
       activateSubDir = 'Scripts'
-      List tempDirParts = script.env.TEMP.split(':')
-      tempDir = "/${tempDirParts[0]}${tempDirParts[1].replace('\\', '/')}"
     }
 
-    this.destDir = "${tempDir}/${script.env.JOB_BASE_NAME}/${script.env.BUILD_NUMBER}/" +
-      python.split('/').last()
-
+    this.destDir = "${script.env.WORKSPACE}/.venv/${python.split('/').last()}"
```

Comment:
This is problematic. If a build on a given node needs 2 different venvs for the same Python version, in the best case there will be an error, in the worst case the existing venv will be reused and some bugs might sneak in.

The problem existed before, but it was countered by the `.cleanup()` method.

Reply:
Let's instead use a random number, similar to what we do in `postgres-utils` for the port. See 5089c97.